tests/components/vallox/test_config_flow.py

```diff
@@ -154,3 +146,115 @@ async def test_form_already_configured(hass: HomeAssistant) -> None:
 
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+async def test_reconfigure_host(hass: HomeAssistant, init_reconfigure_flow) -> None:
+    """Test that the host can be reconfigured."""
+    entry, init_flow_result = init_reconfigure_flow
+
+    reconfigure_result = await hass.config_entries.flow.async_configure(
+        init_flow_result["flow_id"],
+        {
+            "host": "192.168.100.60",
+        },
+    )
+    await hass.async_block_till_done()
+    assert reconfigure_result["type"] is FlowResultType.ABORT
+    assert reconfigure_result["reason"] == "reconfigure_successful"
+
+    # changed entry
+    assert entry.data["host"] == "192.168.100.60"
+
+
+async def test_reconfigure_host_to_same_host_as_another_fails(
+    hass: HomeAssistant, init_reconfigure_flow
+) -> None:
+    """Test that changing host to a host that already exists fails."""
+    entry, init_flow_result = init_reconfigure_flow
+
+    # Create second device
+    create_mock_entry(hass=hass, host="192.168.100.70", name="Vallox 2")
+    await do_setup_vallox_entry(hass=hass, host="192.168.100.70", name="Vallox 2")
+
+    reconfigure_result = await hass.config_entries.flow.async_configure(
+        init_flow_result["flow_id"],
+        {
+            "host": "192.168.100.70",
+        },
+    )
+    await hass.async_block_till_done()
+    assert reconfigure_result["type"] is FlowResultType.ABORT
+    assert reconfigure_result["reason"] == "already_configured"
+
+    # entry not changed
+    assert entry.data["host"] == "192.168.100.50"
+
+
+async def test_reconfigure_host_to_invalid_ip_fails(
```

Comment:
These tests handling exceptions and ending with `FORM` should continue so they either end with `CREATE_ENTRY` or `ABORT`.
Now we don't know if the user can continue and recover from the errors.